dev/WindowsAppRuntime_MSIXInstallFromPath/main.cpp

```diff
@@ -0,0 +1,303 @@
+    {
+        if (m_packageFamilyName.empty())
+        {
+            wchar_t packageFamilyName[PACKAGE_FAMILY_NAME_MAX_LENGTH + 1]{};
+            uint32_t packageFamilyNameLength{ static_cast<uint32_t>(ARRAYSIZE(packageFamilyName)) };
+            THROW_IF_WIN32_ERROR_MSG(::PackageFamilyNameFromFullName(m_packageFullName.c_str(), &packageFamilyNameLength, packageFamilyName), "PackageFullName:%ls", m_packageFullName.c_str());
+            m_packageFamilyName = packageFamilyName;
+        }
+        return m_packageFamilyName.c_str();
+    }
+
+    explicit operator bool() const
+    {
+        return m_packageId != nullptr;
+    }
+
+private:
+    std::wstring m_packageFullName;
+    mutable std::wstring m_packageFamilyName;
+    const PACKAGE_ID * m_packageId{};
+    BYTE m_buffer[sizeof(PACKAGE_ID) + (PACKAGE_NAME_MAX_LENGTH + 1 +
+                                        PACKAGE_RESOURCEID_MAX_LENGTH + 1 +
+                                        PACKAGE_PUBLISHERID_MAX_LENGTH + 1) * sizeof(WCHAR)]{};
+};
+
+void Help()
+{
+    wprintf(L"WindowsAppRuntime_MSIXInstallFromPath [options] <path>\n"
+            L"options:\n"
+            L"  -?, --help = Display help\n"
+            L"          -- = End of options\n"
+            L"where:\n"
+            L"  path = Path where Windows App SDK's MSIX packages can be found\n"
+            L"           => Microsoft.WindowsAppRuntime.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.DDLM.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.Main.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.Singleton.*.msix\n");
+}
+
+HRESULT JustDoIt(PCWSTR path) noexcept try
+{
+    wprintf(L"path: %s\n", path);
+
+    // Install packages
+    auto fpath{ std::filesystem::path(path) };
+    auto inventory{ fpath / L"MSIX.inventory" };
+    std::string lineUtf8;
+    std::ifstream f{ inventory };
+    while (getline(f, lineUtf8))
+    {
+        // Skip blank lines
+        if (lineUtf8.empty())
+        {
+            continue;
+        }
+
+        // Skip comments
+        if (lineUtf8[0] == '#')
+        {
+            continue;
+        }
+
+        // Parse the line
+        auto line{ Microsoft::Utf8::ToUtf16<std::wstring>(lineUtf8.c_str()) };
+        auto offset{ line.find(L'=') };
+        THROW_HR_IF_MSG(E_UNEXPECTED, offset == std::wstring::npos, "line:%s", lineUtf8.c_str());
+        auto filename{ line.substr(0, offset) };
+        THROW_HR_IF_MSG(E_UNEXPECTED, filename.empty(), "line:%s", lineUtf8.c_str());
+        auto packageFullName{ std::wstring(line.c_str() + offset + 1) };
+        THROW_HR_IF_MSG(E_UNEXPECTED, packageFullName.empty(), "line:%s", lineUtf8.c_str());
+
+        AddPackageIfNecessary(path, filename, packageFullName);
+    }
+
+    return 0;
+}
+CATCH_RETURN();
+
+void AddPackageIfNecessary(PCWSTR path, const std::wstring& filename, const std::wstring& packageFullName)
+{
+    wprintf(L"Path: %s\n", path);
```

Comment:
Debugging printfer will go away in the follow up PR after VS invoking this is tested